### PR TITLE
data/fonts: make derivations fixed-outputs (fontforge group)

### PR DESCRIPTION
**Pull request by volth**
- [x] data/fonts: made derivation fixed-output (https://github.com/NixOS/nixpkgs/issues/27754)

### CODE REVIEW

File: pkgs/data/fonts/libertine/default.nix [hunk 1]
```diff
@@ -8,11 +8,14 @@ stdenv.mkDerivation rec {
...
   };
 
-  setSourceRoot = "sourceRoot=`pwd`";
+  sourceRoot = ".";
 
   nativeBuildInputs = [ fontforge ];
 
   buildPhase = ''
+    # <nixpkgs/pkgs/build-support/setup-hooks/set-source-date-epoch-to-latest.sh> set it to mtime of ./env-vars, i.e. the current time
+    export SOURCE_DATE_EPOCH=$(stat -c '%Y' ChangeLog.txt)
```
Comment (volth) [inline pkgs/data/fonts/libertine/default.nix]: Workaround for not yet fixed #28218 (PR #28227)

**Comment by globin** [comment]
Travis failure looks relevant

**Comment by volth** [comment]
That means fontforge is not fully fixed in https://github.com/NixOS/nixpkgs/pull/28212, there are still issues on Darwin :(
Is there a way to try Nix/Darwin without apple hardware? A virtual machine image maybe?

**Comment by volth** [comment]
@globin There are at least three issues:
1. unreliable ```$SOURCE_DATE_EPOCH``` on Darwin https://github.com/NixOS/nixpkgs/pull/29825
2. ```fortforge``` produces slightly different font files on Linux and MacOS (or likely being compiled by gcc and clang): https://github.com/fontforge/fontforge/pull/3134#issuecomment-332359700
3. adding ```find $out -type f -exec md5sum {} \;``` to ```installPhase``` revealed that Travis produces one file less on MacOS than on Linux. I was unable to reproduce it locally, it could be a Travis issue or an issue specific to its environment.

So far, I removed Darwin from the list of supported platforms for this font (anyway it has "linux" in its name :) We can add it back when those issues will be fixed.

**Comment by volth** [comment]
@Mic92

**Pull request merged**